backends/gdx-backends-gwt/src/com/badlogic/gwtref/gen/ReflectionCacheGenerator.java

```diff
@@ -16,31 +16,207 @@
+		this.context = context;
+		this.logger = logger;
 		TypeOracle oracle = context.getTypeOracle();
 		assert (oracle != null);
-		JClassType type = oracle.findType(typeName);
-		if (type == null) {
+		JClassType jType = oracle.findType(typeName);
+		if (jType == null) {
 			logger.log(ERROR, "Couldn't find type '" + typeName + "'");
 			throw new UnableToCompleteException();
 		}
 
-		if (type.isInterface() == null) {
+		if (jType.isInterface() == null) {
 			logger.log(ERROR, "Type '" + typeName + "' must be an interface");
 			throw new UnableToCompleteException();
 		}
 
-		ReflectionCacheSourceCreator source = new ReflectionCacheSourceCreator(logger, context, type);
+		TypeOracle typeOracle = context.getTypeOracle();
+		JPackage[] packages = typeOracle.getPackages();
+		List<JType> types = new ArrayList<>();
+		// gather all types from wanted packages
+		for (JPackage p : packages) {
+			for (JClassType t : p.getTypes()) {
+				gatherTypes(t.getErasedType(), types);
+			}
+		}
+
+		// gather all types from explicitly requested packages
+		try {
+			ConfigurationProperty prop = context.getPropertyOracle().getConfigurationProperty("gdx.reflect.include");
+			for (String s : prop.getValues()) {
+				JClassType type = typeOracle.findType(s);
+				if (type != null)
+					gatherTypes(type.getErasedType(), types);
+			}
+		} catch (BadPropertyValueException e) {
+		}
+
+		gatherTypes(typeOracle.findType("java.util.List").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.util.ArrayList").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.util.HashMap").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.util.Map").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.String").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Boolean").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Byte").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Long").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Character").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Short").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Integer").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Float").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.CharSequence").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Double").getErasedType(), types);
+		gatherTypes(typeOracle.findType("java.lang.Object").getErasedType(), types);
+
+		// sort the types so the generated output will be stable between runs
+		Collections.sort(types, new Comparator<JType>() {
+			public int compare (JType o1, JType o2) {
+				return o1.getQualifiedSourceName().compareTo(o2.getQualifiedSourceName());
+			}
+		});
+
+		List<JType> split;

```

Comment:
Could use a line of background.
`// Split types between two classes as a workaround for the java 65535 const limit. `